compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -523,16 +523,16 @@ class ResolverImpl(
 
     fun resolveAnnotationEntry(ktAnnotationEntry: KtAnnotationEntry): AnnotationDescriptor? {
         bindingTrace.get(BindingContext.ANNOTATION, ktAnnotationEntry)?.let { return it }
-        val containingDeclaration = KtStubbedPsiUtil.getContainingDeclaration(ktAnnotationEntry)
-        return if (containingDeclaration?.let { KtPsiUtil.isLocal(containingDeclaration) } == true) {
-            resolveDeclarationForLocal(containingDeclaration)
-            bindingTrace.get(BindingContext.ANNOTATION, ktAnnotationEntry)
-        } else {
-            ktAnnotationEntry.findLexicalScope().let { scope ->
-                annotationResolver.resolveAnnotationsWithArguments(scope, listOf(ktAnnotationEntry), bindingTrace)
-                bindingTrace.get(BindingContext.ANNOTATION, ktAnnotationEntry)
+        KtStubbedPsiUtil.getContainingDeclaration(ktAnnotationEntry)?.let { containingDeclaration ->
+            if (KtPsiUtil.isLocal(containingDeclaration)) {
+                resolveDeclarationForLocal(containingDeclaration)
+            } else {
+                resolveSession.resolveToDescriptor(containingDeclaration).annotations.forEach {}
```

Comment:
What is this empty `forEach()` for? Is it to make the annotation entry to be looked up in binding context?

Reply:
yes, it is similar to `forceResolveAll()` but less expensive.